pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -306,21 +331,37 @@ func (is *InstanceService) InstanceCreate(clusterName string, name string, confi
 					networkID: netID,
 				})
 			}
-		} else if net.UUID == "" && net.SubnetID != "" {
-			subnet, err := subnets.Get(is.networkClient, net.SubnetID).Extract()
-			if err != nil {
-				return nil, err
+		} else if net.Subnets != nil {
+			for _, snet := range net.Subnets {
+				if snet.UUID != "" {
+					subnet, err := subnets.Get(is.networkClient, snet.UUID).Extract()
+					if err != nil {
+						return nil, err
+					}
+					nets = append(nets, ServerNetwork{
+						networkID: subnet.NetworkID,
+						subnetID:  snet.UUID,
+					})
+				} else {
+					sopts := subnets.ListOpts(snet.Filter)
```

Comment:
... alse withouth this else for sure...
do
```
sopts.ID = snet.UUID
sopts.NetworkID = net.UUID
```

Reply:
a-ha! Good catch. I like it :+1: